Source/missiles.cpp

```diff
@@ -883,9 +886,8 @@ bool MonsterTrapHit(int monsterId, int mindam, int maxdam, int dist, missile_id
 		MonsterDeath(monster, monster.direction, true);
 	} else if (resist) {
 		PlayEffect(monster, 1);
-	} else {
-		if (monster.type().type != MT_GOLEM)
-			M_StartHit(monster, dam);
+	} else if (!monster.isPlayerOwned()) {

```

Comment:
Same as the case for MonsterMHit, this allows non-player golems to be interrupted by trap hits.

Reply:
This is because golems cannot be interupted, not because the player is controlling it.
```suggestion
	} else if (monster.type().type != MT_GOLEM) {
```